### PR TITLE
feat(positionService): add timeout to legacy subgraphs

### PR DESCRIPTION
**Pull request by th0rOdinson**
- With this change:
   - All promises, except the last version, will timeout after 30 seconds.
   - If a promise times out or encounters an error, it will resolve with { data: null, error: true }.

### CODE REVIEW

File: src/services/positionService.ts [hunk 1]
```diff
@@ -163,8 +163,16 @@ export default class PositionService {
         );
       })
     );
-
-    const results = await Promise.all(promises.map((promise) => promise.catch(() => ({ data: null, error: true }))));
+    const results = await Promise.all(
+      promises.map(async (promise, index) => {
+        const isLastVersion = networksAndVersions[index].version === POSITIONS_VERSIONS[POSITIONS_VERSIONS.length - 1];
```
Comment (FiboApe) [inline src/services/positionService.ts:168]: There is a constant that you can use `LATEST_VERSION`

File: src/services/positionService.ts [hunk 2]
```diff
@@ -163,8 +163,16 @@ export default class PositionService {
...
+        try {
+          return await timeoutPromise(promise, !isLastVersion ? '30s' : undefined);
+        } catch {
+          return { data: null, error: true };
```
Comment (FiboApe) [inline src/services/positionService.ts:172]: should we return a promise too? `Promise.resolve({ data: null, error: true })`

File: src/services/positionService.ts (hunk 1 above)
Comment (th0rOdinson) [reply src/services/positionService.ts:168]: Awesome! Thanks

File: src/services/positionService.ts (hunk 2 above)
Comment (th0rOdinson) [reply src/services/positionService.ts:172]: I researched the benefits of adding `Promise.resolve()`, and in this case, using `async` ensures the function always returns a promise. However, I can add it for the sake of code consistency.

Comment (FiboApe) [reply src/services/positionService.ts:172]: Nah, its gucci as it is!

**Review by FiboApe — APPROVED** [review]

**Pull request merged**